lib/make/makefile.bacmet

```diff
@@ -0,0 +1,209 @@
+DIAMOND_DB_PATH = .
+
+# Specify this when you want to point to a specific, non-generic, database.
+# Set DIAMOND_DB in a local Makefile to the
+# full path of the database file, without the ".dmnd" suffix.
+DIAMOND_DB =
+
+# deeparg.py specific macro: define minimum a.a. alignemnt length threshold
+ALENGTH = 37.5
+
+# Set this to suffixes corresponding to the types of files you wish to fetch with some
+# of the t$(DIAMOND_DB)ets.
+ACCEPTED_SUFFIXES =
+
+# DIAMOND_ALIGN_OPTS sets non-essential parameters for diamond alignment
+#
+# See the Diamond documentation available at GitHub:
+# https://github.com/bbuchfink/diamond. 
+#
+# Look particularly at options for sensitivity, scoring and memory and
+# performance.
+#
+# To set options, override this by inserting a row similar to the one below but
+# with whatever options you'd like to add in your Makefile *after* you include
+# this file.
+DIAMOND_ALIGN_OPTS =  
+
+# Output format of the DIAMOND results. The default value refers to a tabular file.
+# For more format options see https://github.com/bbuchfink/diamond.
+OUTFMT = 6
+
+# Download HTTP URL for the BacMet protein database
+# In case you are not using ftp, you can override this macro and paste here the https URL.
+HTTP = http://bacmet.biomedicine.gu.se/download/BacMet2_EXP_database.fasta
+# IMPORTANT: databases are updated recurrently, so always check the availability of the http
+
+# Download HTTPS URL for the bacmet.py script, necessary for the downstream analyses
+HTTPS = https://gist.github.com/08ab264f0ea127cc242d99bc3844c743.git 
+# IMPORTANT: bacmet.py needs to be placed inside the directory that has the DIAMOND output
+# so run mirror_bacmet.py into the directory that contains .bacmet.sort.best.m8 files
+# N.B.: the python script "bacmetg.py" requires python 3.6*-3.8*
+
+# *** Internal ***
+
+# MAKECALL_DIAMOND_BACMET is a macro that defines what will be output to the .makecall
+# file, the file that records versions, file stamps, parameters etc.
+#
+# *Don't redefine!*
+MAKECALL_VERSION        = echo "`date +"%Y-%m-%d %H:%M:%S"`: $@ was made with `diamond --version`" > $@.makecall
+MAKECALL_PARAMS         = echo "	Called with parameters: $(DIAMOND_ALIGN_OPTS)" >> $@.makecall
+MAKECALL_INFILES        = echo "	Input files: $^ (`ls -lL $^|tr '\n' ','`)" >> $@.makecall
+MAKECALL_DB_BACMET      = echo "	Database: $(DIAMOND_DB_PATH)/$(DIAMOND_DB) (`ls -lL $(DIAMOND_DB_PATH)/$(DIAMOND_DB).dmnd`)" >> $@.makecall
+MAKECALL_DIAMOND_BACMET = $(MAKECALL_VERSION); $(MAKECALL_PARAMS); $(MAKECALL_DB_BACMET); $(MAKECALL_INFILES)
+MAKECALL_MAKEDB         = $(MAKECALL_VERSION); $(MAKECALL_INFILES)
+
+# *** misc targets ***
+
+# compress all .fastq files in the work directory.
+compress_all_fastqs: $(subst .fastq,.fastq.gz, $(wildcard *.fastq))
+
+# Interleave two fastq files (named %.R1.fastq and %.R2.fastq respectively).
+#
+# The target uses a python script -- interleave_fastq.py -- cloned from this
```

Comment:
We're using `mergepe` from the `seqtk` package for this in https://github.com/LNUc-EEMiS/metatdenovo/tree/dev (check out `modules/nf-core/modules/seqtk/mergepe/main.nf`). More robust than a Python 2.7 script in a gist.

Reply:
Interesting! I proposed `interleave_fastq.py` as it was mentioned in `makefile.misc`.
Since the target is already present in `makefile.misc`, hence being redundant, I was planning on removing it anyways.